### PR TITLE
+rem remote deployment whitelist feature

### PR DESCRIPTION
**Pull request by ktoso**
Initial working version, will do a bit of cleanup soon.

### CODE REVIEW

**Comment by ktoso** [comment]
Resolves #21663

**Comment by ktoso** [comment]
Oh and needs docs.

File: akka-remote/src/test/scala/akka/remote/RemoteDeploymentWhitelistSpec.scala [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    muteSystem(system)
+    remoteSystem.eventStream.publish(TestEvent.Mute(
+      EventFilter[EndpointException](),
+      EventFilter.error(start = "AssociationError"),
+      EventFilter.warning(pattern = "received dead letter.*(InboundPayload|Disassociate|HandleListener)")))
+  }
+
+  private def byteStringOfSize(size: Int) = ByteString.fromArray(Array.fill(size)(42: Byte))
+
+  val maxPayloadBytes = system.settings.config.getBytes("akka.remote.test.maximum-payload-bytes").toInt
```
Comment (patriknw) [inline akka-remote/src/test/scala/akka/remote/RemoteDeploymentWhitelistSpec.scala]: cleanup the test, copy pasta from something else now, I think

**Comment by ktoso** [comment]
Cleaned up

**Comment by ktoso** [comment]
Added docs (and addressed missing docs in foldAsync/scanAsync - let me smuggle those through in this PR...)

remoting_ _akka_documentation

**Comment by patriknw** [comment]
wat, do you have a quota on number of PRs?

**Comment by ktoso** [comment]
Heh ok, splitting it out 😉

**Comment by ktoso** [comment]
Split out to: https://github.com/akka/akka/pull/21701

**Comment by ktoso** [comment]
Can I merge this?

File: akka-docs/rst/java/remoting.rst [hunk 2]
```diff
@@ -168,6 +168,23 @@ you can advise the system to create a child on that remote node like so:
 
 .. includecode:: code/docs/remoting/RemoteDeploymentDocTest.java#deploy
 
+Remote deployment whitelist
+---------------------------
+
+As remote deployment can potentially be abused aby both users and even attackers a whitelist feature
```
Comment (patriknw) [inline akka-docs/rst/java/remoting.rst]: s/aby/by/

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by drewhk** [comment]
LGTM

**Pull request merged**